### DISCUSSION TITLE
Parser for Xit files in JavaScript

### DISCUSSION THREAD

**Discussion by DLvalentine** [jotaen/xit#48] [Show & tell]
Howdy all!!

First off, thanks so much @jotaen - I *love* Xit and it has become my favorite productivity thing. It captures in plaintext what my process was already more-or-less like for managing my life lol.

At any rate, I'm working on a light editor for Xit files for desktop/web/mobile, and started by writing a really simple parser in JS to be used for that. It is far from perfect, and I imagine I'll be revising it a lot (it was my first time writing a parser for anything, really) but I thought it good enough to share:

https://github.com/DLvalentine/xit-parse

https://www.npmjs.com/package/xit-parse

I hope someone finds it useful! I'll circle back to this discussion and share my editor if/when I finish it :) 

The README is a little lacking at the moment, and ideally it would have installation/usage/examples, but I'll get that nailed down soon enough :)

**Comment by jotaen** [comment]
Cool, thanks for sharing this! 🎉 

I’m happy to link your parser from the [[x]it! website](https://xit.jotaen.net/) if you like (so that other developers can discover it more easily). I’d just need to think about how to fit it in there, since it’s more like a developer’s library rather than an end-user tool, so maybe there could be a “libraries” section or so.

**Reply by DLvalentine** [reply]
Absolutely! If it makes sense and you're happy to do that, I'd be over the moon!

I have noticed a few minor things with it that I intend to patch up soon, but I want to test drive it in the xit editor I'm building to see if I can't flush out anything else that might need fixing. 

e.g. the way I export the module now, its usage is a little clunky, but I have a plan to shore it up. It could also use some better documentation :D
